### PR TITLE
💁🏻  Move`shared/` to `server/public`

### PR DESCRIPTION
**Pull request by aileen**
refs #8221

Instead of serving our shared assets from a `shared/` folder, we' move everything, that is used to `server/public`

### CODE REVIEW

File: core/server/middleware/serve-favicon.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ function serveFavicon() {
                     res.end(content.body);
                 });
             } else {
-                filePath = path.join(config.get('paths:corePath'), 'shared', 'favicon.ico');
+                filePath = path.join(config.get('paths:corePath'), 'server/public', 'favicon.ico');
```
Comment (AileenCGN) [inline core/server/middleware/serve-favicon.js]: Would it make sense actually, to have a new config path property for the public assets here? e. g. `assetPath`?

Comment (ErisDS) [inline core/server/middleware/serve-favicon.js]: Seems like a good idea to me

File: core/server/middleware/serve-shared-file.js [hunk 2]
```diff
@@ -8,12 +8,13 @@ var crypto = require('crypto'),
 // Handles requests to robots.txt and favicon.ico (and caches them)
 function serveSharedFile(file, type, maxAge) {
```
Comment (ErisDS) [inline core/server/middleware/serve-shared-file.js]: Can we rename this middleware as well please?

File: core/server/middleware/serve-shared-file.js [hunk 3]
```diff
@@ -8,12 +8,13 @@ var crypto = require('crypto'),
...
     var content,
-        corePath = config.get('paths').corePath,
+        corePath = path.join(config.get('paths').corePath, '/server'),
         filePath,
         blogRegex = /(\{\{blog-url\}\})/g,
         apiRegex = /(\{\{api-url\}\})/g;
+    console.log('file:', file);
```
Comment (ErisDS) [inline core/server/middleware/serve-shared-file.js]: I know this is WIP, but I saw this so thought I'd mention it ;)

Comment (AileenCGN) [inline core/server/middleware/serve-shared-file.js]: Hehe... Yup

File: core/server/middleware/serve-shared-file.js (hunk 2 above)
Comment (AileenCGN) [inline core/server/middleware/serve-shared-file.js]: I was thinking about that, but not sure.. Would `servePublicFile` be good? Oh... renaming...

Comment (ErisDS) [inline core/server/middleware/serve-shared-file.js]: Yes exactly

**Comment by AileenCGN** [comment]
@ErisDS this is ready for review and merging 🎉

File: core/server/middleware/serve-public-file.js [hunk 4]
```diff
@@ -4,18 +4,18 @@ var crypto = require('crypto'),
...
+function servePublicFile(file, type, maxAge) {
     var content,
-        corePath = config.get('paths').corePath,
+        publicFilePath = config.get('paths').publicFilePath,
         filePath,
         blogRegex = /(\{\{blog-url\}\})/g,
         apiRegex = /(\{\{api-url\}\})/g;
 
-    filePath = file.match(/^shared/) ? path.join(corePath, file) : path.join(corePath, 'shared', file);
+    filePath = file.match(/^public/) ? path.join(publicFilePath, file.replace(/^public/, '')) : path.join(publicFilePath, file);
```
Comment (ErisDS) [inline core/server/middleware/serve-public-file.js]: Let's remove this line, and also the `public/` from the paths in blog/app.js and make this consistent? All the assets are now neatly colocated, let's keep the code clean?

Comment (AileenCGN) [inline core/server/middleware/serve-public-file.js]: Makes sense!

**Comment by kirrg001** [comment]
For future reference. This is the reason why we have separated assets files for Ghost+Ghost-Admin:
https://github.com/TryGhost/Ghost/pull/8343#issuecomment-301445159

**Pull request merged**